### PR TITLE
Initial check-in of open AI search app from .Net to Python

### PR DESCRIPTION
**Pull request by dm-chelupati**
this is same as sample - https://github.com/Azure-Samples/azure-functions-openai-aisearch-dotnet/tree/main?tab=readme-ov-file#azure-functions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,131 @@
...
+<!--
+---
+name: Azure Function using OpenAI trigger and bindings extension to highlight OpenAI retrieval augmented generation with Azure AI Search.
+description: This repository contains an Azure Function using OpenAI trigger and bindings extension to highlight OpenAI retrieval augmented generation with Azure AI Search. The sample uses managed identity.
+page_type: sample
+products:
+- azure-functions
+- azure
+- entra-id
+urlFragment: azure-functions-openai-aisearch-dotnet8
```
Comment (paulyuk) [inline README.md]:
```suggestion
urlFragment: azure-functions-openai-aisearch-python
```

Need to pythonize all the metadata like this

File: README.md [hunk 2]
```diff
@@ -1 +1,131 @@
...
+## Using Azure Functions OpenAI trigger and bindings extension to import data and query with Azure Open AI and Azure AI Search
+
+This sample contains an Azure Function using OpenAI bindings extension to highlight OpenAI retrieval augmented generation with Azure AI Search.
+
+You can learn more about the OpenAI trigger and bindings extension in the [GitHub documentation](https://github.com/Azure/azure-functions-openai-extension) and in the [Official OpenAI extension documentation](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-openai)
+
+
+## Prerequisites
+
+* [.NET 8 SDK](https://dotnet.microsoft.com/download/dotnet/6.0) or greater (Visual Studio 2022 recommended)
```
Comment (paulyuk) [inline README.md]: Please steal prereqs from another sample like this one:

https://github.com/Azure-Samples/function-python-ai-openai-chatgpt

File: README.md [hunk 3]
```diff
@@ -1 +1,131 @@
...
+
+If you don't run azd provision, you can create an [OpenAI resource](https://portal.azure.com/#create/Microsoft.CognitiveServicesOpenAI) and an [AI Search resource](https://portal.azure.com/#create/Microsoft.Search) in the Azure portal to get your endpoints. After it deploys, click Go to resource and view the Endpoint value.  You will also need to deploy a model, e.g. with name `chat` with model `gpt-35-turbo` and `embeddings` with model `text-embedding-3-small`
+
+### Create local.settings.json (Should be in the same folder as host.json. Automatically created if you ran azd provision)
+```json
+{
+  "IsEncrypted": false,
+  "Values": {
+    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
+    "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated",
```
Comment (paulyuk) [inline README.md]:
```suggestion
    "FUNCTIONS_WORKER_RUNTIME": "python",
```

File: README.md [hunk 4]
```diff
@@ -1 +1,131 @@
...
+
+## Run your app using Visual Studio Code
+
+1. Open the folder in a new terminal.
+1. Run the `code .` code command to open the project in Visual Studio Code.
+1. In the command palette (F1), type `Azurite: Start`, which enables debugging without warnings.
+1. Press **Run/Debug (F5)** to run in the debugger. Select **Debug anyway** if prompted about local emulator not running.
+1. Send POST requests to the `ingest` and `ask` endpoints respectively using your HTTP test tool. If you have the [RestClient](https://marketplace.visualstudio.com/items?itemName=humao.rest-client) extension installed, you can execute requests directly from the [`test.http`](./app/test.http) project file.
+
+## Run your app using Visual Studio
```
Comment (paulyuk) [inline README.md]: Please cut entire Visual Studio section.  It's not really a strong client for python

File: infra/core/host/appserviceplan.bicep [hunk 5]
```diff
@@ -0,0 +1,20 @@
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+
+param kind string = ''
+param reserved bool = true
+param sku object
+
+resource appServicePlan 'Microsoft.Web/serverfarms@2023-12-01' = {
```
Comment (paulyuk) [inline infra/core/host/appserviceplan.bicep:9]: consider bumping these versions to latest supported in 2024

File: infra/core/host/functions-flexconsumption.bicep
Comment (paulyuk) [inline infra/core/host/functions-flexconsumption.bicep:1]: Nice!  consider taking the newest functions-flexconsumption.bicep i posted in chat for review.  It is more dynamic between system and user assigned identity

File: infra/scripts/createlocalsettings.ps1 [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        if ($line -match "AZURE_OPENAI_ENDPOINT"){
+            $OpenAIEndPoint = ($line -split "=")[1] -replace '"',''
+        }
+    }
+
+    @{
+        "IsEncrypted" = "false";
+        "Values" = @{
+            "AzureWebJobsStorage" = "UseDevelopmentStorage=true";
+            "FUNCTIONS_WORKER_RUNTIME" = "dotnet-isolated";
```
Comment (paulyuk) [inline infra/scripts/createlocalsettings.ps1:21]:
```suggestion
            "FUNCTIONS_WORKER_RUNTIME" = "python";
```

File: infra/scripts/createlocalsettings.sh [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+            OpenAIEndPoint=$(echo "$line" | cut -d '=' -f 2 | tr -d '"')
+        fi
+    done <<< "$output"
+
+    cat <<EOF > ./app/local.settings.json
+{
+    "IsEncrypted": "false",
+    "Values": {
+        "AzureWebJobsStorage": "UseDevelopmentStorage=true",
+        "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated",
```
Comment (paulyuk) [inline infra/scripts/createlocalsettings.sh:28]:
```suggestion
        "FUNCTIONS_WORKER_RUNTIME": "python",
```

**Review by paulyuk — CHANGES_REQUESTED** [review]
Thanks a lot @dm-chelupati!

Just a few comments for you.

File: infra/scripts/createlocalsettings.sh (hunk 7 above)
Comment (dm-chelupati) [reply infra/scripts/createlocalsettings.sh:28]: fixed

File: infra/scripts/createlocalsettings.ps1 (hunk 6 above)
Comment (dm-chelupati) [reply infra/scripts/createlocalsettings.ps1:21]: fixed

File: infra/core/host/appserviceplan.bicep (hunk 5 above)
Comment (dm-chelupati) [reply infra/core/host/appserviceplan.bicep:9]: fixed

File: README.md (hunk 4 above)
Comment (dm-chelupati) [reply README.md]: fixed

File: infra/core/host/functions-flexconsumption.bicep
Comment (dm-chelupati) [reply infra/core/host/functions-flexconsumption.bicep:1]: I didnt see difference between your code vs my code in this bicep file.

File: README.md (hunk 3 above)
Comment (dm-chelupati) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (dm-chelupati) [reply README.md]: fixed

File: README.md (hunk 1 above)
Comment (dm-chelupati) [reply README.md]: fixed

**Pull request closed without merging**